dask/dataframe/tests/test_io.py

```diff
@@ -697,25 +697,37 @@ def test_to_hdf():
         out = pd.read_hdf(fn, '/data')
         tm.assert_frame_equal(df, out[:])
 
-    # saving to multiple datasets
+def test_to_hdf_multiple_datasets():
+    df = pd.DataFrame({'x': ['a', 'b', 'c', 'd'],
+                       'y': [1, 2, 3, 4]}, index=[1., 2., 3., 4.])
     a = dd.from_pandas(df, 2)
+    df16 = pd.DataFrame({'x': ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p'],
+                       'y': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]},
+                            index=[1., 2., 3., 4., 5., 6., 7., 8., 9., 10., 11., 12., 13., 14., 15., 16.])
+    b = dd.from_pandas(df16, 16)
+
+    # saving to multiple datasets making sure order is kept
+    with tmpfile('h5') as fn:
+        b.to_hdf(fn, '/data*')
+        out = dd.read_hdf(fn, '/data*')
+        tm.assert_frame_equal(df16, out.compute())
```

Comment:
I recommend the `eq` function present in this module.  It will check some other handy things

```python
eq(df16, out)
```